package/MDAnalysis/coordinates/XYZ.py

```diff
@@ -408,7 +409,12 @@ def Writer(self, filename, n_atoms=None, **kwargs):
         See Also
         --------
         :class: `XYZWriter`
+
         """
+        if isinstance(atoms, AtomGroup):
```

Comment:
That would allow an API which I'm not comfortable with since it allows to use an integer or ANY other type which implements `__len__`. I rather have an interface with typehints so that we can warn users if stuff happens that might have unintentional side effects.